src/segment-loader.js

```diff
@@ -667,17 +675,15 @@ export default videojs.extend(videojs.EventTarget, {
                                                    this.sourceUpdater_.buffered());
 
     // Update segment meta-data (duration and end-point) based on timeline
-    let timelineUpdated = updateSegmentMetadata(playlist,
-                                                currentMediaIndex,
-                                                timelineUpdate);
+    timelineUpdated = updateSegmentMetadata(playlist, currentMediaIndex, timelineUpdate);
 
     // the last segment append must have been entirely in the
     // already buffered time ranges. adjust the timestamp offset to
     // fetch forward until we find a segment that adds to the
     // buffered time ranges and improves subsequent media index
     // calculations.
-//    if (!timelineUpdated) {
-//      this.expired_ -= segment.duration;
-//    }
+    if (!timelineUpdated) {
```

Comment:
Might not be strictly necessary, but it seemed to read a little better. That said, maybe that isn't true. I'll try it out without the variable and see how it looks.